### PR TITLE
[77] add feature in change job's status API @PUT

### PR DESCRIPTION
**Pull request by SuitedIceZ**
## What did you do
- endpoint ```GET /profiles/:id``` add a feature that the same role can't view each other
- endpoint ```PUT /jobs/:id/status```
- add ForbiddenException when change JobStatus is not in order
- change ApplicationStatus of application in the job when changing JobStatus
image

## Demo
- https://dev.modela.miello.dev/
- login with any casting in one window
- login with actor that has resumes (maybe 1,2) in another window
- as casting create new job
- as actor apply that created job
- as casting change that JobStatus in order until "เสร็จสิ้นการคัดเลือก" (SELECTION_END)
- any pending actors will automatically change ApplicationStatus to "ไม่ผ่านการคัดเลือก" (REJECTED)
- also admin cancell job will change to REJECTED

## Limitation
- not have unit test yet

### CODE REVIEW

File: apps/api/src/modules/user/profile/profile.service.ts
Comment (miello) [inline apps/api/src/modules/user/profile/profile.service.ts:66]:
```suggestion
    const targetUser = await this.repository.getUserProfileById(id)
```

File: apps/api/src/modules/user/profile/profile.service.ts
Comment (miello) [inline apps/api/src/modules/user/profile/profile.service.ts:77]:
```suggestion
    if (user.userId !== id && user.type === TargetUser.type) {
        // user can't see other user's profile if they are the same type
        throw new ForbiddenException()
    }
```

File: apps/api/src/modules/job/job.service.ts
Comment (miello) [inline apps/api/src/modules/job/job.service.ts:419]:
```suggestion
    if (
      this.JobStatusToNumber(job.status) + 1 !==
        this.JobStatusToNumber(updateJobStatus) &&
      updateJobStatus !== JobStatus.CANCELLED
    )
```

**Review by miello — CHANGES_REQUESTED** [review]

**Review by miello — COMMENTED** [review]
LGTM, please wait for review from @palmcm

**Review by palmcm — APPROVED** [review]
LGTM

**Review by miello — APPROVED** [review]

**Pull request merged**
